### PR TITLE
Upgrade to rustc 1.42.0-nightly (9b98af84c 2019-12-22)

### PR DESCRIPTION
**Pull request by SimonSapin**

### CODE REVIEW

File: components/config/Cargo.toml [hunk 1]
```diff
@@ -30,5 +30,5 @@ servo_config_plugins = { path = "../config_plugins" }
 [dev-dependencies]
 std_test_override = { path = "../std_test_override" }
 
-[target.'cfg(not(any(target_os = "android", feature = "uwp")))'.dependencies]
```
Comment (SimonSapin) [inline components/config/Cargo.toml]: This caused a new warning:

```
warning: /home/simon/servo1/components/config/Cargo.toml: Found `feature = ...` in `target.'cfg(...)'.dependencies`. This key is not supported for selecting dependencies and will not work as expected. Use the [features] section instead: https://doc.rust-lang.org/cargo/reference/manifest.html#the-features-section
```

As far as I know it’s not possible to make a Cargo feature disable a dependency.

File: components/net/http_cache.rs [hunk 2]
```diff
@@ -230,12 +230,15 @@ fn get_response_expiry(response: &Response) -> Duration {
         let max_heuristic = Duration::hours(24) - age;
         let heuristic_freshness = if let Some(last_modified) =
             // If the response has a Last-Modified header field,
-            // caches are encouraged to use a heuristic expiration value
-            // that is no more than some fraction of the interval since that time.
-            response.headers.typed_get::<LastModified>() {
+        // caches are encouraged to use a heuristic expiration value
+        // that is no more than some fraction of the interval since that time.
+            response.headers.typed_get::<LastModified>()
```
Comment (CYBAI) [inline components/net/http_cache.rs]: Per https://github.com/servo/servo/pull/25301#discussion_r358068316, I filed rust-lang/rustfmt#3979 👀

**Pull request merged**